### PR TITLE
feat: allow Chrome-specific params in user contexts

### PR DESCRIPTION
**Pull request by OrKoN**
This PR adds a python proxy server wrapper over a Node.js proxy server to test the URLs.

### CODE REVIEW

File: src/bidiMapper/domains/browser/BrowserProcessor.ts [hunk 1]
```diff
@@ -38,9 +40,20 @@ export class BrowserProcessor {
     return {};
   }
 
-  async createUserContext(): Promise<Browser.CreateUserContextResult> {
+  async createUserContext(
+    params: Record<string, any>
```
Comment (Lightning00Blade) [inline src/bidiMapper/domains/browser/BrowserProcessor.ts:44]: Consider using something similar to [this TypeScript ](https://www.typescriptlang.org/play?#code/LAKAlgdgLgpgTgMwIYGMYAIDCcZNgITgHsB3AZ3kyOhgA8oAlGARwFcYyp0BvUddAPQAqIX35D0ASQToyABxgowCMDAAmAGnRqw8ohTLooAC13oU1WPXQljMCNpgAjVgHNXkV7I5kw1bboWEBCKUGQAdGLoQgJiOnoUAPIQACIwUKjGAPwAXOhOREQANrgQANxiwqIg-NHoAArEtACe3nAAbvBavgC2YEVIcEZERnbo1BhySGQUasPoALQLck3NCxQd8FExYitELQDK8J1wubJQcJ4VNYIi2w2r+c1TM+hFulDdYH0DQ1AjJgwE3QL1m8yWexaCycz2mZAW70421iN0hzXwsJmABkPmdOJcIK5rvwqvcAIIOIhyKB+CBIIpvD7jGRES4eCCGf7oVxwJDQdCsCDvPqwOYoYgzBassDs9CoNCvf6RG51eqDMLM0YYACqDCx40Bf2MfNG+gwQU4YCgrBphPGbMgcpw6BlEFZ6mVtR2N2l7LIAHUrcZtRAwCcyPSAHLpEisgDWZJQCrIeIungA2gBda4AX1AoDoclZXAtXFchVcjRgKlo6AAvOgAOTloiuHKNuWGUvXAu0ItwLhQZ4YAAqRAA4hWAGr09hkAA8I4AfPWeGJ0wBpF0OOMwZpEGQjzvoLd0WAQNSGfGedBZdAAAwAJNwhwoD9yK1Waznnxuc-f0DyEITkzPIR03bNQBzHsQFfDBYE4Vcx0nVsZyKOd52wXACGIchKEsM8mDYDgoCXGDSyMPIEK4BteBudNmwrHJ4iLJJUnSTJG1Aow4HYDQoJgqAYNAIA)?

File: tools/http-proxy.mjs [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        );
+        proxyResponse.pipe(originalResponse, {end: true});
+      }
+    );
+
+    originalRequest.pipe(proxyRequest, {end: true});
+  })
+  .listen();
+
+console.log(
```
Comment (Lightning00Blade) [inline tools/http-proxy.mjs:48]: Nit: Let's prefix the log messages, as in `Requested: <X>`, `Server started on: <Y>`

**Review by Lightning00Blade — APPROVED** [review]

File: tests/browser/test_create_user_context.py [hunk 3]
```diff
@@ -52,3 +52,40 @@ def user_context(x):
...
                   }],
                                               key=user_context)
+
+
+@pytest.mark.asyncio
+async def test_browser_create_user_context_proxy_server(
+        websocket, http_proxy_server):
+
+    # Localhost URLs are not proxied.
+    example_url = "http://example.com"
```
Comment (sadym-chromium) [inline tests/browser/test_create_user_context.py:62]: What about `127.0.0.1`?

**Review by sadym-chromium — APPROVED** [review]

File: tests/browser/test_create_user_context.py (hunk 3 above)
Comment (OrKoN) [reply tests/browser/test_create_user_context.py:62]: it's the same. I think it skips proxy for anything that is pointing to local IP.

Comment (OrKoN) [reply tests/browser/test_create_user_context.py:62]: It works for `192.x.x.x` IPs but we could figure out how to compute those later

File: tools/http-proxy.mjs (hunk 2 above)
Comment (OrKoN) [reply tools/http-proxy.mjs:48]: It would require more parsing on the python side.

**Pull request merged**